core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -241,6 +249,16 @@ public void writeRawBytes(byte[] value, int offset, int length) throws IOExcepti
       encoder.writeBytes(value, offset, length);
    }
 
+   @Override
+   public TagWriter subWriter(int number, boolean nested) throws IOException {
+      Encoder subEncoder = encoder.subEncoder(number);
```

Comment:
@wburns you can have the max nested depth here:
```suggestion
      Encoder subEncoder = encoder.subEncoder(number, serCtx.getConfiguration().maxNestedMessageDepth());
```

Reply:
Actually, this is not where I wanted to add it, let me look into this a bit more again now that I have time :)

Reply:
Ended up just implementing as an additional argument as I couldn't get the other ways to work the way I wanted.